packages/Webkul/Checkout/src/Cart.php

```diff
@@ -294,11 +294,13 @@ public function getItemByProduct($data)
         $items = $this->getCart()->all_items;
 
         foreach ($items as $item) {
-            if ($item->product->getTypeInstance()->compareOptions($item->additional, $data['additional'])) {
-                if (isset($data['additional']['parent_id'])) {
-                    if ($item->parent->product->getTypeInstance()->compareOptions($item->parent->additional, request()->all())) {
-                        return $item;
-                    }
+            if ($item->product->getTypeInstance()->compareOptions($item->additional, $data['additional']) 
+                && 
+                isset($data['additional']['parent_id']) 
+                && 
+                $item->parent->product->getTypeInstance()->compareOptions($item->parent->additional, request()->all())
+               ) {                
+                    return $item;                    
```

Comment:
The cases in the master version are added more precisely to return void also. But  in your case it will always return an item.